runtime/src/iree/base/internal/math.h

```diff
@@ -288,7 +288,7 @@ static inline float iree_math_f16_to_f32(const uint16_t f16_value) {
   }
   const uint32_t u32_value = sign | exp | mantissa;
   float f32_value;
-  memcpy(&f32_value, &u32_value, sizeof(f32_value));
+  memcpy(&f32_value, &u32_value, sizeof f32_value);
```

Comment:
Is there a particular reason for these changes to how `sizeof` is used? I see 1077 results for `\ssizeof\(` (`sizeof(foo)`) and only 44 results for `\ssizeof\s` (`sizeof foo`) in the runtime/ folder.

Reply:
It's just that the parentheses are extraneous and really just dropped in the evaluation of this expression, so that they are slightly misleading in terms of what `sizeof` is here. They make it look like a function-like keyword, as in `sizeof(type)`. But here, `f32_value` is not a type, it is a variable, so here `sizeof` really is an operator.

I think it's worth not obscuring the fact that `sizeof` is an operator. Often, the reason why expressions that involve `sizeof` work as intended is that `sizeof` has sufficiently high operator precedence. For example, `sizeof(x) + sizeof(y)` returns the expected value because `sizeof` has higher precedence than `+`.

Reply:
This reminds me of a fight I got into with my C language professor on the first day of college. His whole introduction to C was based around "C only has one builtin function (sizeof)". He was really mad at me when I told him that in fact it has zero because sizeof is an operator. He wouldn't back down no matter what references I showed him, said it was not good enough proof. I think it was just because he was teaching out of the book he wrote, and it was wrong in there.

I wouldn't say that my college experience really *improved* from that point.

Reply:
Lol :-)

Reply:
Good to know, thanks. Could be helpful to change the style throughout the project (if it's worth it here, it should be worth it in other places too).